src/ledger/LedgerManagerImpl.cpp

```diff
@@ -345,6 +345,7 @@ LedgerManagerImpl::loadLastKnownLedger(function<void()> handler)
                     {
                         mApp.getBucketManager().assumeState(
                             has, header.current().ledgerVersion);
+                        mApp.getBucketManager().initAllIndexes();
```

Comment:
Initially, we did have that function in `assumeState`. However, in several codepaths, `assumeState` gets called as part of a work task. This would cause bucket indexing to take a very long time on startup since the task would keep getting slept. By moving it to `loadLastKnownLedger`, we can index synchronously and it's a lot faster. I originally did not index in bucket application for the same reason. I'll have to investigate it a little more, but I don't seem to be failing anything without indexing during bucket application but I want to look into it.

Reply:
I'm not sure I'm following how this would work without rebuilding indexes after bucket application: for example, in case of online catchup, we start core at LCL=1 then replay buckets to arrive at some ledger N, while buffering ledgers from the network. `loadLastKnownLedger` only  covers the case when we start core with all the buckets already in place, but this isn't always the case, right?